akka-actor/src/main/scala/akka/actor/AbstractProps.scala

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.actor
+
+import java.lang.reflect.{ Modifier, ParameterizedType, TypeVariable }
+
+import akka.dispatch._
+import akka.japi.Creator
+import akka.routing._
+import akka.util.Reflect
+
+import scala.annotation.varargs
+import scala.language.existentials
+import scala.reflect.ClassTag
+import akka.japi.Util.immutableSeq
+
+/**
+ * Java API: Factory for Props instances.
+ */
+private[akka] trait AbstractProps {
+
+  def validate(clazz: Class[_]) =
```

Comment:
note that it should also have the INTERNAL API marker in the Scala doc comment, the reason is that it will be public from Java